staging/src/k8s.io/apiextensions-apiserver/test/integration/table_test.go

```diff
@@ -0,0 +1,183 @@
+
+	scheme := runtime.NewScheme()
+	codecs := serializer.NewCodecFactory(scheme)
+	parameterCodec := runtime.NewParameterCodec(scheme)
+	metav1.AddToGroupVersion(scheme, gv)
+	scheme.AddKnownTypes(gv, &metav1beta1.Table{}, &metav1beta1.TableOptions{})
+	scheme.AddKnownTypes(metav1beta1.SchemeGroupVersion, &metav1beta1.Table{}, &metav1beta1.TableOptions{})
+
+	crConfig := *config
+	crConfig.GroupVersion = &gv
+	crConfig.APIPath = "/apis"
+	crConfig.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: codecs}
+	crRestClient, err := rest.RESTClientFor(&crConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ret, err := crRestClient.Get().
+		Resource(crd.Spec.Names.Plural).
+		SetHeader("Accept", fmt.Sprintf("application/json;as=Table;v=%s;g=%s, application/json", metav1beta1.SchemeGroupVersion.Version, metav1beta1.GroupName)).
+		VersionedParams(&metav1beta1.TableOptions{}, parameterCodec).
+		Do().
+		Get()
+	if err != nil {
+		t.Fatalf("failed to list %v resources: %v", gvk, err)
+	}
+
+	tbl, ok := ret.(*metav1beta1.Table)
+	if !ok {
+		t.Fatalf("expected metav1beta1.Table, got %T", ret)
+	}
+	t.Logf("%v table list: %#v", gvk, tbl)
+
+	if got, expected := len(tbl.ColumnDefinitions), 5; got != expected {
+		t.Errorf("expected %d headers, got %d", expected, got)
+	} else {
+		alpha := metav1beta1.TableColumnDefinition{Name: "Alpha", Type: "string", Format: "", Description: "Custom resource definition column (in JSONPath format): .spec.alpha", Priority: 0}
+		if got, expected := tbl.ColumnDefinitions[2], alpha; got != expected {
+			t.Errorf("expected column definition %#v, got %#v", expected, got)
+		}
+
+		beta := metav1beta1.TableColumnDefinition{Name: "Beta", Type: "integer", Format: "int64", Description: "the beta field", Priority: 42}
+		if got, expected := tbl.ColumnDefinitions[3], beta; got != expected {
+			t.Errorf("expected column definition %#v, got %#v", expected, got)
+		}
+
+		gamma := metav1beta1.TableColumnDefinition{Name: "Gamma", Type: "integer", Description: "a column with wrongly typed values"}
+		if got, expected := tbl.ColumnDefinitions[4], gamma; got != expected {
+			t.Errorf("expected column definition %#v, got %#v", expected, got)
+		}
+	}
+	if got, expected := len(tbl.Rows), 1; got != expected {
+		t.Errorf("expected %d rows, got %d", expected, got)
+	} else if got, expected := len(tbl.Rows[0].Cells), 5; got != expected {
+		t.Errorf("expected %d cells, got %d", expected, got)
+	} else {
+		if got, expected := tbl.Rows[0].Cells[0], "foo"; got != expected {
+			t.Errorf("expected cell[0] to equal %q, got %q", expected, got)
+		}
+		if got, expected := tbl.Rows[0].Cells[2], "foo_123"; got != expected {
+			t.Errorf("expected cell[2] to equal %q, got %q", expected, got)
+		}
+		if got, expected := tbl.Rows[0].Cells[3], int64(10); got != expected {
+			t.Errorf("expected cell[3] to equal %#v, got %#v", expected, got)
+		}
+		if got, expected := tbl.Rows[0].Cells[4], "bar"; got != expected {
```

Comment:
compare this behaviour: the cells are not rejected (we don't have a way in the table types to return an error, skipping and object feels wrong), but the wrong type is just passed along.

Do we want that? If not, what do we want instead?

Reply:
I would say we should pass `<error>` as a string for now.  We can always change later (output is not stable).

Reply:
Actually, let's just pass nil for that particular cell.  Nil cells should always be allowed.

Reply:
done